ibm/service/mqcloud/utils.go

```diff
@@ -0,0 +1,196 @@
+package mqcloud
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"strconv"
+
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/conns"
+	"github.com/IBM-Cloud/terraform-provider-ibm/ibm/flex"
+	rc "github.com/IBM/platform-services-go-sdk/resourcecontrollerv2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+
+	"github.com/IBM/mqcloud-go-sdk/mqcloudv1"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func loadWaitForQmStatusEnvVar() bool {
+	envValue := os.Getenv("IBMCLOUD_MQCLOUD_WAIT_FOR_QM_STATUS")
+	return strings.ToLower(envValue) != "false"
+}
+
+var waitForQmStatus = loadWaitForQmStatusEnvVar()
+var planCache = make(map[string]string)
+
+const qmStatus = "running"
+const qmCreating = "initializing"
+const isQueueManagerDeleting = "true"
+const isQueueManagerDeleteDone = "true"
+const reservedDeploymentPlan = "reserved-deployment"
+const enforceReservedDeploymentPlan = true
+
+// waitForQmStatusUpdate Waits for QM to be in running state
+func waitForQmStatusUpdate(context context.Context, d *schema.ResourceData, meta interface{}) (interface{}, error) {
+	mqcloudClient, err := meta.(conns.ClientSession).MqcloudV1()
+	if err != nil {
+		return "", err
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{qmCreating},
+		Target:  []string{qmStatus},
+		Refresh: func() (interface{}, string, error) {
+			getQueueManagerStatusOptions := &mqcloudv1.GetQueueManagerStatusOptions{}
+			parts, err := flex.SepIdParts(d.Id(), "/")
+			if err != nil {
+				return nil, "", err
+			}
+			getQueueManagerStatusOptions.SetServiceInstanceGuid(parts[0])
+			getQueueManagerStatusOptions.SetQueueManagerID(parts[1])
+			queueManagerStatus, response, err := mqcloudClient.GetQueueManagerStatusWithContext(context, getQueueManagerStatusOptions)
+			if err != nil {
+				return "", "", fmt.Errorf("GetQueueManagerWithContext ...... %s err: %s", response, err)
+			}
+			if queueManagerStatus == nil || queueManagerStatus.Status == nil {
+				return nil, "", fmt.Errorf("queueManagerStatus or queueManagerStatus.Status is nil")
+			}
+			fmt.Println("The queue manager is currently in the " + *queueManagerStatus.Status + " state ....")
```

Comment:
Can we remove this fmt stmt i think its not required Terraform will log it still creating...